libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,645 @@
+	 * contains output and divider
+	 */
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(&asciiStr[2], holder, sizeof(holder));
+
+	holder = strtok(NULL, dividers);
+	/* Adding here 1 because always is taken next place after the point of divide */
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[sizeof(asciiStr) / 2 + 1], holder);
+	/*
+	 * Starting from 2 because of the last strtok cut of the first two elements for token
+	 * contains output and divider
+	 */
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(&asciiStr[sizeof(asciiStr) / 2 + 1], holder, sizeof(holder));
+
+	holder = strtok(NULL, dividers);
+	/*
+	 * After getting everything available from the string
+	 * pointer will be set as 0 and the value of the next reads will be NULL
+	 */
+	TEST_ASSERT_EQUAL_PTR(NULL, holder);
+	TEST_ASSERT_NULL(holder);
+}
+
+TEST(string_tok, null_char_impact)
+{
+	char holder[ASCII_LENGTH];
+
+	create_ascii_set(1);
+	asciiStr[ASCII_LENGTH / 2] = '\0';
+
+	/* Last element of ascii string is \127 and between 0 and 127 was written 0*/
+	memcpy(holder, strtok(asciiStr, "\127"), ASCII_LENGTH);
+	create_ascii_set(1);
+
+	/* \0 was placed in half of the ASCII set. This case is looking for sensitivity for \0 in strings*/
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(holder, asciiStr, ASCII_LENGTH / 2);
+	TEST_ASSERT_NOT_EQUAL_INT(0, memcmp(holder, asciiStr, ASCII_LENGTH));
+}
+
+
+TEST(string_tok, empty_input_output)
+{
+	char empty[] = { 0 };
+
+	/* Checking if strtok will read all elements in substring and another his search will return null*/
+	TEST_ASSERT_NOT_NULL(strtok("abc", "d"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, "d"));
+
+	/* Different scenarios of acquiring null pointer as output*/
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok("", "d"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, ""));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok("", "d"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, "\0"));
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(empty, ""));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, "\0"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(empty, ""));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, ""));
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(empty, "\0"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, "\0"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(empty, "\0"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, ""));
+}
+
+
+TEST(string_tok, call_to_call_sep)
+{
+	char loremStr[20] = LOREM_IPSUM,
+		 separator[2] = { 0 };
+	int i;
+
+
+	/*
+	 * In this case we checking if is possible to
+	 * pass different stop points in each call to
+	 * get desired tokens
+	 */
+	TEST_ASSERT_EQUAL_PTR(loremStr, strtok(loremStr, "I"));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[7], strtok(NULL, "m"));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[11], strtok(NULL, "D"));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[13], strtok(NULL, "r"));
+
+	/*
+	 * After using strtok on some elements we must
+	 * again write it to the array if we want to use
+	 * the same set because strtok breaks the elements in the string
+	 */
```

Comment:
Tip for the future, you can leave it now. We want to have concise comments, so in this case if you could write for example `strtok() changes the content of a string, that's why we have to set it once again`.